### PR TITLE
oath: code -s gave wrong error message

### PR DESCRIPTION
**Pull request by dagheyman**
Found this by accident, but it prints the multiple hits error when there is no hits.

### CODE REVIEW

**Comment by dagheyman** [comment]
> Shouldn't it still exit with nonzero status code if there is no match?

I'm not sure, can't you see it as the command completed successfully even if there was no match?

**Comment by emlun** [comment]
If you're setting the `-s` option then you presumably want to get exactly one match, so I do think zero matches should be regarded a failure. :slightly_smiling_face:

**Comment by dagheyman** [comment]
> If you're setting the `-s` option then you presumably want to get exactly one match, so I do think zero matches should be regarded a failure. slightly_smiling_face

Fair enough.

File: ykman/cli/oath.py [hunk 1]
```diff
@@ -368,10 +368,13 @@ def code(ctx, show_hidden, query, single):
             if e.sw == SW.SECURITY_CONDITION_NOT_SATISFIED:
                 ctx.fail('Touch credential timed out!')
 
-    elif single:
+    elif single and len(creds) > 1:
         _error_multiple_hits(ctx, [cr for cr, c in creds])
 
-    if single:
+    elif single and len(creds) == 0:
+        ctx.fail('No credential found.')
```
Comment (emlun) [inline ykman/cli/oath.py]:
```suggestion
        ctx.fail('No matching credential found.')
```

Comment (emlun) [reply ykman/cli/oath.py]: Maybe?

**Pull request merged**